### DISCUSSION TITLE
Does anyone have a rbi file for Technicolor DGA0122/ DGA0122NLK or know how to restore from a raw bank dump?

### DISCUSSION THREAD

**Discussion by protonbeam** [hack-technicolor/hack-technicolor#218] [Q&A]
> If there's anyone with UNO uk can you please download and upload the rbi file to the repository from this link: https://help.uno.uk/knowledge-base/article/technicolor-dwa0120 you can only download from within their network. they have other technicolor firmwares aswell.

_Originally posted by @kingpeppermint in https://github.com/hack-technicolor/hack-technicolor/issues/189#issuecomment-1022727912_

Does anyone have a rbi file for Technicolor DGA0122 like above or is there a guide of how to restore 
from a raw bank dump?
https://hack-technicolor.readthedocs.io/en/stable/Repository/#dwa0122-vcnt-2

Thanks

**Comment by LuKePicci** [comment]
Unfortunately nobody from uno.uk shared the RBI, the bank dump is restorable but you need to be root to write the raw image (or use a nand writer on desoldered chip)
________________________________
From: protonbeam ***@***.***>
Sent: Wednesday, March 2, 2022 4:31:35 PM
To: hack-technicolor/hack-technicolor ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [hack-technicolor/hack-technicolor] Does anyone have a rbi file for Technicolor DGA0122/ DGA0122NLK or know how to restore from a raw bank dump? (Issue #217)


If there's anyone with UNO uk can you please download and upload the rbi file to the repository from this link: https://help.uno.uk/knowledge-base/article/technicolor-dwa0120 you can only download from within their network. they have other technicolor firmwares aswell.

Originally posted by @kingpeppermint<https://github.com/kingpeppermint> in #189 (comment)<https://github.com/hack-technicolor/hack-technicolor/issues/189#issuecomment-1022727912>

Does anyone have a rbi file for Technicolor DGA0122 or is there a guide of how to restore
from a raw bank dump?
https://hack-technicolor.readthedocs.io/en/stable/Repository/#dwa0122-vcnt-2

Thanks

**Comment by protonbeam** [comment]
Thanks for the reply - how do you write as root if the router won't boot I'm thinking of trying to change the booted bank but its hard to find any information on the Technicolor DGA0122 (there isn't even a full manual) on how to do this - do you have any ideas?   

I'm basically trying to find a way to fix what went wrong after I attempted to install tch-nginx-gui = why did it Erase bank_1 and does this router have a bank_2 which can be copied to bank_1.
- https://github.com/Ansuel/tch-nginx-gui/issues/1130 
```
- `root@dsldevice:~# /etc/init.d/rootdevice force
passwd: password for root changed by root
Unlocking /dev/mtd4 ...

Writing from /dev/mtd3 to /dev/mtd4 ...
Unlocking bank_1 ...
Erasing bank_1 ...
Rebooting ...`
```
Thanks

**Comment by protonbeam** [comment]
This video may help give a better idea of what happens when it boots - https://user-images.githubusercontent.com/100783100/156420019-f2deff13-587f-47d6-b7b2-521f5784f5da.mp4

**Comment by kingpeppermint** [comment]
> This video may help give a better idea of what happens when it boots - https://user-images.githubusercontent.com/100783100/156420019-f2deff13-587f-47d6-b7b2-521f5784f5da.mp4

It looks like it's stuck in bootP or in other words, a technicolor router has two banks who in turn have a firmware and a backup firmware, it's failing to boot either.  If you would have an RBI file you can recover it by setting up an dhcp server and offering the RBI file. If not the only option is soldering (opening the router casing, de-soldering the chip and programming it with external tools(the nand writer) as @LuKePicci referred to.

**Comment by protonbeam** [comment]
Thanks for the reply 

so it would try bank_2 before moving onto bootP - does this imply bank_2 is not bootable either?

Does failboot has a chance to still work - (https://hack-technicolor.readthedocs.io/en/stable/Resources/#crazy-power-switching) or is still best to get a serial Adapter to see what the router is doing.

Is there anywhere you can find a rbi file for Technicolor DGA0122 or can create your own?

It looks like I'm not the only 1 this has happened to https://forums.whirlpool.net.au/archive/9xv6j74j maybe we need a don't install tch-nginx-gui on this device list especially it there isn't a widely available rbi file. 

Are there any guides for de-soldering the chip and programming it from a bank dump?

Thanks

**Comment by LuKePicci** [comment]
> so it would try bank_2 before moving onto bootP - does this imply bank_2 is not bootable either?
Exactly. The source of your troubles is the fact that you installed the tch-nginx-gui which automatically implements the "optimal bank plan" as described in out bank planning guide step, where you might have noticed that notable exception warning about the "Missing RBI".

Your current situation is a normal "easy to recover" condition for people who has their board's RBI.

**Reply by protonbeam** [reply]
So does the tch-nginx-gui install itself in bank_2 then Erases bank_1 (https://github.com/Ansuel/tch-nginx-gui/issues/1130#issue-1157321464) ?  

I must have missed the  "optimal bank plan" on https://github.com/Ansuel/tch-nginx-gui  just out of curiosity where did you see this? 

Has anyone managed to get a copy of a RBI that would work on a technicolor DGA0122 or tried asking technicolor for any other RBI especially as I suspect some of the code in the RBI must Open Source licensed? https://www.technicolor.com/contact/products-support for it?

Thanks

**Reply by LuKePicci** [reply]
> So does the tch-nginx-gui install itself in bank_2 then Erases bank_1?

Yes, it installs itself into the overlay of bank_2 (not bank_2 partition, that one contains the firmware) and erases bank_1 partition (after that it also uses the bank_1 partition for modoverlay but this is another story). Overlay of bank_1 is kept clean.

**Reply by FrancYescO** [reply]
> Has anyone managed to get a copy of a RBI that would work on a technicolor DGA0122 or tried asking technicolor for any other RBI especially as I suspect some of the code in the RBI must Open Source licensed?

RBI for ISP-distributed devices can be provided only by ISP, Technicolor probably will also have no access to these compiled firmwares distributed by ISPs

you can ask technicolor for an RBI of a directly distributed devices, but idk if any of this ever existed

**Reply by kingpeppermint** [reply]
I contacted Technicolor, they indeed say they don't provide those files. i tried to get them to get me trough someone with SSE who can provide it but that seemed impossible too, it did do something because i got a response from SSE, but still no file.

I highly doubt that there is any difference between the rbi from SSE and the rbi from Uno other then maybe the wifi name. But if someone ever is able to supply the Uno one, we can at least try, the router will reject it anyway if it wouldn't work and even if it would brick it, that's exactly what it is now already realistically the case 
 
Open source, no they don't need to provide the firmware on that ground because they have a page explaining which software open source software they use and the version numbers, but the firmware also contains files(mainly the technicolor branding) who are not open source, so they don't supply the full image because the license does not permit that. gotta love loopholes right?

**Reply by FrancYescO** [reply]
Any RBI file that match your board can be potentially be flashed on your device via TFTP.

So yes, you "just" need to get an RBI from any ISP that is designed for you device.

**Reply by protonbeam** [reply]
> I contacted Technicolor, they indeed say they don't provide those files. i tried to get them to get me trough someone with SSE who can provide it but that seemed impossible too, it did do something because i got a response from SSE, but still no file.
> 
> I highly doubt that there is any difference between the rbi from SSE and the rbi from Uno other then maybe the wifi name. But if someone ever is able to supply the Uno one, we can at least try, the router will reject it anyway if it wouldn't work and even if it would brick it, that's exactly what it is now already realistically the case
> 
> Open source, no they don't need to provide the firmware on that ground because they have a page explaining which software open source software they use and the version numbers, but the firmware also contains files(mainly the technicolor branding) who are not open source, so they don't supply the full image because the license does not permit that. gotta love loopholes right?

Does anyone know if a RBI exist if there has never being a firmware update? How were the RBI files for the other router on this site obtained?

Has anyone tried emailing Uno to see if they would be happy to email it in the context of reducing e-waste? 

We also have:
Andrews & Arnold https://support.aa.net.uk/Category:Technicolor_DGA0122, 
1pbroadband - https://1pbroadband.com/index.taf?cprefs=agree

**Reply by Whitewizarddonut** [reply]
I can provide a rbi file and upload a unlocked firmware.
You will have to avoid auto updates that dms does sometimes.

However. For this I want firmware I can remote upload to a device and screen + file grab. 

I have admin access to telstras system. 
Basically I would have to upload a unlocked version remotely.

**Reply by kingpeppermint** [reply]
hi, that would be VERY welcome, in here they say https://hack-technicolor.readthedocs.io/en/stable/Repository/#what-to-do-if-your-firmware-is-not-listed-here you can open an new issue to upload it, if that's not what you looking for wait for @LuKePicci to reply as a believe he is currently managing this project. (otherwise i assume he can correct me if i'm wrong and point you to the right direction)

**Comment by kingpeppermint** [comment]
can you put together the download url for the rbi file using this file: 
`config single_config 'cwmpd_config'
	option state '0'
	option upgradesmanaged '1'
	option periodicinform_enable '1'
	option periodicinform_interval '43200'
	option periodicinform_time '0'
	option connectionrequest_port '51007'
	option ssl_castore '/etc/ssl/certs/'
	option ssl_certificate '/etc/ssl/tch_cwmp_ssl.cert'
	option ssl_privatekey '/etc/ssl/tch_cwmp_ssl.key'
	option ssl_hostnamecheck '1'
	option ssl_datecheck '0'
	option ssl_verifypeer '1'
	option acs_url 'http://nld-acs.com/'
	option enforce_https '0'
	option connectionrequest_auth '0'
	option connectionrequest_allowedips '0.0.0.0/0,::/0'
	option connectionrequest_throttle_time '60'
	option connectionrequest_throttle_number '15'
	option interface 'wan'
	option interface6 'wan6'
	option ip_preference 'v4_only'
	option upgrade_switchovertype '0'
	option upgrade_rollback_timeout '0'
	option rollback_cwmpd '1'
	option rollback_delayed '1'
	option rollback_gui '0'
	option rollback_unknown '0'
	option namespace_version '1-2'
	option backoff_minwait '5'
	option backoff_multiplier '2000'
	option activenotification_throttle '0'
	option encrypt_passwords '0'
	list forcedactives 'InternetGatewayDevice.Layer3Forwarding.DefaultConnectionService'`

Cwmpd in /etc/config in the raw dump from the reporsority site.
I'm fine with scanning the url but i don't not how the figure out the port and such because directly contacting nld-acs doesn't work.

**Comment by kingpeppermint** [comment]
So i found out Uno, the provider offering the rbi files, is on the Talktalk wholesale network llu-tbb. So if there's anyone with talktalk or on the talk talk network, can you give it a try please?

**Reply by Whitewizarddonut** [reply]
I have telstras rbi files

**Comment by Whitewizarddonut** [comment]
It doesn't work that way. 
You need to change the policy on the device to be unlocked. 
I have all the copies of the firmware, however they check daily if it matches the system policy and if not it schedules a automatic update. 
It's part of the way that telstra is combating the spam calls. 
I can provide the latest firmware, but from my own playing around in the system. You will either need to be put on the unlocked version or just disable the code that searches for the update. 
You won't ever be able to truly lock telstra out thou as it's easy to access and enable remote admin for basically any device from inside the network. 
They do a daily sweep, and have stat's on how many people using what. 
Sent from Yahoo Mail on Android 
 
  On Thu, 1 Sept 2022 at 8:25 pm, ***@***.***> wrote:   


hi, that would be VERY welcome, in here they say https://hack-technicolor.readthedocs.io/en/stable/Repository/#what-to-do-if-your-firmware-is-not-listed-here you can open an new issue to upload it, if that's not what you looking for wait for @LuKePicci to replay as a believe he is currently managing this project. (otherwise i assume he can correct me if i'm wrong and point you to the right direction)

**Reply by kingpeppermint** [reply]
Yes, after recovering(where we need the rbi files for) and (re-)rooting an device one of the first steps is ''harderning root acces'' which includes removing all links to the  calling home function.

**Comment by Whitewizarddonut** [comment]
https://ibb.co/D4MTFFChttps://ibb.co/YNhrQ0220220902-104654  
|  
|   
|   
|   |    |

   |

  |
|  
|   |  
20220902-104654
 
Image 20220902-104654 hosted in ImgBB
  |   |

  |

  |

  
This is how its managed. 
I can provide the files. But i want help with applyi g the git myself. 
I have root access to my own modem. Am still learning python and how to use kali properly. 
But which of these files will help
Sent from Yahoo Mail on Android 
 
  On Fri, 2 Sept 2022 at 8:18 am, ***@***.***> wrote:   


Yes, after recovering(where we need the rbi files for) and (re-)rooting an device one of the first steps is ''harderning root acces'' which includes removing all links to the calling home function.
